### PR TITLE
fix(code-snippet): use Carbon focus outline and reposition scroll bar

### LINKED ISSUE #2304: Focus outline on Code Snippet is unstyled
When clicking in a code snippet, it appears that we have a un-styled focus indicator:

image

This is called (I believe) by `tabindex="0"` and `role="textbox"`, which I believe is moreso meant for an area where you can freely edit text.

### LINKED ISSUE #5219: React: Code Snippet single line text shifted upward
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

When using the `CodeSnippet` component in single line mode, a horizontal scroll bar will render, which shifts the `<pre>` wrapped code upward so that it is not centered.

**Behavior observed in [Storybook](http://react.carbondesignsystem.com/?path=/story/codesnippet--single-line)**
Screen Shot 2020-01-30 at 2 02 10 PM

> Is this issue related to a specific component?

This is related to `CodeSnippet` when the `type` prop is set to `single`.

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I expected to see the code centered as it is when the text width does not exceed the container. Instead, the text shifted upwards, possibly offset by the scrollbar.

> What browser are you working in?

This behavior was observed in:
- Chrome 79.0.3945.130
- Safari 13.0.3 (15608.3.10.1.4)
- Firefox 72.0.1

> What version of the Carbon Design System are you using?

This is Carbon 10 design system, with the `7.9.0` version of `carbon-components-react`

## Steps to reproduce the issue

1. Create a `<CodeSnippet type="single" />` component
2. Provide child text that is long enough to exceed the width of the component
3. Observe a horizontal scroll bar render and offset the Y position of the text so that it is no longer centered.

> Please create a reduced test case in CodeSandbox

Reproducible demo here: https://codesandbox.io/s/musing-surf-1jyot

Additionally, this is observable in the public Storybook example: http://react.carbondesignsystem.com/?path=/story/codesnippet--single-line

### PR DESCRIPTION
**Pull request by emyarod**
Closes #2304
Closes #5219

This PR adds Carbon focus styles to the single line code snippet instead of using the browser default. The component is also repositioned to account for the scroll bar across different platforms

#### Testing / Reviewing

Ensure that the focus style is visually correct, and that scroll bars look correct on all platforms (including using various scroll bar preferences in macOS System Preferences > General)

### CODE REVIEW

File: packages/components/src/components/code-snippet/_code-snippet.scss [hunk 1]
```diff
@@ -97,16 +97,21 @@
     max-width: rem(760px);
     min-width: rem(320px);
     height: $carbon--spacing-08;
-    padding: 0 56px 0 $carbon--spacing-05;
+    padding: 0 carbon--mini-units(5) 0 0;
```
Comment (tw15egan) [inline packages/components/src/components/code-snippet/_code-snippet.scss]: Any reason to not use `$carbon--spacing-08` here?

**Review by tw15egan — APPROVED** [review]
Tiny nit, other than that it looks great in Chrome / Firefox / Safari with or without scrollbars always enabled.

**Review by laurenmrice — APPROVED** [review]
looks good to me ! 🙌🏻 thanks for figuring this out

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

**Comment by asudoh** [comment]
Got a green light from @emyarod to mark this as "status: ready to merge".

**Pull request merged**